### PR TITLE
Migrates "opennlp-sandbox" from JUnit 4 to JUnit5

### PR DESCRIPTION
**Pull request by rzo1**
- applies bulk migration from JUnit 4 to JUnit 5
- applies formatting according to OpenNLP style

### CODE REVIEW

**Review by kinow — COMMENTED** [review]
@rzo1 ,

```
 Error:  Errors: 
Error:    GradientDescentUtilsTest.testConvergence:34 » Runtime failed to converge at it...
[INFO] 
Error:  Tests run: 34, Failures: 0, Errors: 1, Skipped: 2
```

**Comment by rzo1** [comment]
@kinow yes. This is one of the toggling Tests, which appears from time to time. Also with JUnit 4  and on Unix/OSX + Windows from time to time 🫠

File: corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCasWriterTest.java [hunk 1]
```diff
@@ -53,21 +54,19 @@ public static void setUp() throws IOException {
...
       fail(e.getLocalizedMessage());
     }
-    new TestCorpusServer();
   }
 
   @Test
-  @Ignore
-  // TODO Investigate why this test fails with:
-  //  A CasConsumer descriptor specified implementation class "org.apache.opennlp.corpus_server.connector.CSCasWriter",
-  //  but this class does not implement the CasConsumer interface.
```
Comment (mawiesne) [inline corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCasWriterTest.java:69]: This comment should not be lost. Please re-add the 2nd part of the comment.

File: corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCollectionReaderTest.java [hunk 2]
```diff
@@ -82,14 +83,12 @@ public static void setUp() throws IOException {
       e.printStackTrace();
       fail(e.getLocalizedMessage());
     }
-    new TestCorpusServer();
   }
 
   @Test
-  @Ignore
-  // TODO Investigate why this test fails with:
-  //  No value has been assigned to the mandatory configuration parameter corpusName.
```
Comment (mawiesne) [inline corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCollectionReaderTest.java:91]: This comment should not be lost. Please re-add that part of the comment.

File: opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/MultiSentenceSearchResultsProcessorTest.java [hunk 3]
```diff
@@ -20,21 +20,21 @@
...
 
-	public class MultiSentenceSearchResultsProcessorTest {
+class MultiSentenceSearchResultsProcessorTest {
 
-	private final MultiSentenceSearchResultsProcessor proc = new MultiSentenceSearchResultsProcessor();
+  private final MultiSentenceSearchResultsProcessor proc = new MultiSentenceSearchResultsProcessor();
 
-	@Test
-	@Ignore
-	// TODO OPENNLP-1455 This test case fails with: "UnknownHostException: api.datamarket.azure.com: nodename nor servname provided, or not known"
```
Comment (mawiesne) [inline opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/MultiSentenceSearchResultsProcessorTest.java:34]: This JIRA-issue related comment should not be lost. Please re-add it.

File: opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/StoryDiscourseNavigatorTest.java [hunk 4]
```diff
@@ -18,22 +18,21 @@
...
 
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public class StoryDiscourseNavigatorTest {
+class StoryDiscourseNavigatorTest {
 
-	@Test
-	@Ignore
-	// TODO OPENNLP-1455 This test fails with "UnknownHostException: api.datamarket.azure.com: nodename nor servname provided, or not known"
```
Comment (mawiesne) [inline opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/StoryDiscourseNavigatorTest.java:32]: This JIRA-issue related comment should not be lost. Please re-add it.

File: opennlp-similarity/src/test/java/opennlp/tools/similarity/apps/taxo_builder/TaxonomyBuildMatchTest.java [hunk 5]
```diff
@@ -16,37 +16,37 @@
...
   @Test
-  public void testTaxonomySeedImport() {
+  void testTaxonomySeedImport() {
     AriAdapter ad = new AriAdapter();
     ad.getChainsFromARIfile("src/test/resources/taxonomies/irs_dom.ari");
     assertTrue(ad.lemma_AssocWords.size() > 0);
   }
 
   @Test
-  @Ignore // TODO Check if this test works as expected, and: why it takes so much time.
```
Comment (mawiesne) [inline opennlp-similarity/src/test/java/opennlp/tools/similarity/apps/taxo_builder/TaxonomyBuildMatchTest.java:34]: This comment should not be lost. Please re-add it.

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/LeskEvaluatorTest.java [hunk 6]
```diff
@@ -21,23 +21,23 @@
...
 
 import opennlp.tools.disambiguator.datareader.SensevalReader;
 
-public class LeskEvaluatorTest extends AbstractEvaluatorTest {
+class LeskEvaluatorTest extends AbstractEvaluatorTest {
 
   static SensevalReader seReader = new SensevalReader();
 
   @Test
-  @Ignore // TODO OPENNLP-1446: Investigate why test fails while parsing 'EnglishLS.train'
```
Comment (mawiesne) [inline opennlp-wsd/src/test/java/opennlp/tools/disambiguator/LeskEvaluatorTest.java:34]: This JIRA-issue related comment should not be lost. Please re-add it.

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/MFSEvaluatorTest.java [hunk 7]
```diff
@@ -21,18 +21,18 @@
...
 
 import opennlp.tools.disambiguator.datareader.SensevalReader;
 
-public class MFSEvaluatorTest extends AbstractEvaluatorTest {
+class MFSEvaluatorTest extends AbstractEvaluatorTest {
 
   static SensevalReader seReader = new SensevalReader();
 
   @Test
-  @Ignore // TODO OPENNLP-1446: Investigate why test fails while parsing 'EnglishLS.train'
```
Comment (mawiesne) [inline opennlp-wsd/src/test/java/opennlp/tools/disambiguator/MFSEvaluatorTest.java:34]: This JIRA-issue related comment should not be lost. Please re-add it.

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java [hunk 8]
```diff
@@ -46,10 +46,10 @@ public class WSDEvaluatorTest extends AbstractEvaluatorTest {
   static WSDModel model;
 
   static ArrayList<String> testWords;
-  
+
   @Test
-  @Ignore // TODO OPENNLP-1446: Investigate why test fails while parsing 'EnglishLS.train'
```
Comment (mawiesne) [inline opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java:51]: This JIRA-issue related comment should not be lost. Please re-add it.

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java [hunk 9]
```diff
@@ -98,8 +98,8 @@ public void testTraining() {
   }
 
   @Test
-  @Ignore // Make this work once we have migrated to JUnit5 in the sandbox components
```
Comment (mawiesne) [inline opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java:101]: This comment should not be lost. Please re-add it.

File: tf-ner-poc/src/test/java/org/apache/opennlp/namefinder/PredictTest.java [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
 import java.io.IOException;
 import java.nio.file.Path;
 
-public class PredictTest {
+class PredictTest {
 
-  @Test @Ignore
-  // TODO This test is not platform neutral and, for instance, fails with:
-  //  "Cannot find TensorFlow native library for OS: darwin, architecture: aarch64"
-  //  We need JUnit 5 in the sandbox to circumvent this, so it can be run in supported environments
```
Comment (mawiesne) [inline tf-ner-poc/src/test/java/org/apache/opennlp/namefinder/PredictTest.java:16]: This comment should not be lost. Please re-add it.

**Review by mawiesne — CHANGES_REQUESTED** [review]

File: corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCollectionReaderTest.java (hunk 2 above)
Comment (rzo1) [reply corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCollectionReaderTest.java:91]: done

File: corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCasWriterTest.java (hunk 1 above)
Comment (rzo1) [reply corpus-server/corpus-server-connector/src/test/java/org/apache/opennlp/corpus_server/connector/CSCasWriterTest.java:69]: done

File: opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/MultiSentenceSearchResultsProcessorTest.java (hunk 3 above)
Comment (rzo1) [reply opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/MultiSentenceSearchResultsProcessorTest.java:34]: done

File: opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/StoryDiscourseNavigatorTest.java (hunk 4 above)
Comment (rzo1) [reply opennlp-similarity/src/test/java/opennlp/tools/parse_thicket/apps/StoryDiscourseNavigatorTest.java:32]: done

File: opennlp-similarity/src/test/java/opennlp/tools/similarity/apps/taxo_builder/TaxonomyBuildMatchTest.java (hunk 5 above)
Comment (rzo1) [reply opennlp-similarity/src/test/java/opennlp/tools/similarity/apps/taxo_builder/TaxonomyBuildMatchTest.java:34]: done

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/MFSEvaluatorTest.java (hunk 7 above)
Comment (rzo1) [reply opennlp-wsd/src/test/java/opennlp/tools/disambiguator/MFSEvaluatorTest.java:34]: done

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java (hunk 8 above)
Comment (rzo1) [reply opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java:51]: done

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java (hunk 9 above)
Comment (rzo1) [reply opennlp-wsd/src/test/java/opennlp/tools/disambiguator/WSDEvaluatorTest.java:101]: done

File: tf-ner-poc/src/test/java/org/apache/opennlp/namefinder/PredictTest.java (hunk 10 above)
Comment (rzo1) [reply tf-ner-poc/src/test/java/org/apache/opennlp/namefinder/PredictTest.java:16]: done

File: opennlp-wsd/src/test/java/opennlp/tools/disambiguator/LeskEvaluatorTest.java (hunk 6 above)
Comment (rzo1) [reply opennlp-wsd/src/test/java/opennlp/tools/disambiguator/LeskEvaluatorTest.java:34]: done

**Comment by rzo1** [comment]
Didn't expect the migration tool to kill comments, so good catch @mawiesne - they should be restored now.

**Review by kinow — APPROVED** [review]

**Pull request merged**